source/common/secret/secret_manager_impl.cc

```diff
@@ -86,13 +87,26 @@ TlsCertificateConfigProviderSharedPtr SecretManagerImpl::findOrCreateTlsCertific
       removeDynamicSecretProvider(map_key);
     };
 
-    secret_provider = std::make_shared<TlsCertificateSdsApi>(
+    secret_provider = create_fn(unregister_secret_provider);
+    dynamic_secret_providers_[map_key] = secret_provider;
+  }
+  return secret_provider;
+}
+
+TlsCertificateConfigProviderSharedPtr SecretManagerImpl::findOrCreateTlsCertificateProvider(
+    const envoy::api::v2::core::ConfigSource& sds_config_source, const std::string& config_name,
+    Server::Configuration::TransportSocketFactoryContext& secret_provider_context) {
+  std::function<SdsApiSharedPtr(std::function<void()> unregister_secret_provider)> create_fn =
```

Comment:
just use auto as
auto create_fn = [&]

in the capture, not need to list all used variable,  just one & to tell compiler that all used variables are using reference.

Reply:
Done.